tests/unit/Service/SQS.service.test.js

```diff
@@ -129,14 +138,35 @@ describe('Service/SQS', () => {
       });
     });
 
-    it('throws an error if publish fails', async () => {
-      const service = getService({
-        sendMessage: new Error('SQS is down!'),
-      }, false);
+    [
+      SQS_PUBLISH_FAILURE_MODES.CATCH,
+      '',
+      undefined,
+      'another-value',
```

Comment:
I'm not sure I agree with these cases. If some random value is passed in, that's a programmer error. If we want to use a value for something else in the future, it shouldn't be providing a certain behaviour in this version.

Reply:
60a7217d2cb5c5bd8ed0ce8568ec5cc02fe9fd0c

Reply:
I was thinking more
```
if (!Object.values(SQS_PUBLISH_FAILURE_MODES).includes(failureMode)) {
  throw new Error(`bad value for failureMode: ${failureMode}`);
}
```

I don't think `''` and `undefined` should be allowed values. We've got a default provided in the arguments list so they don't need to be accepted for backwards compatibility or anything.

Reply:
91f9403fc92d8f686f35ff21b813e54d0b7913a0